### ISSUE TITLE
MaximumWeightBipartiteMatching provides wrong result

### ISSUE THREAD

**Issue by mjblackhorse** [jgrapht/jgrapht#1026]
```
 * JGraphT version: I checked 1.2 and 1.4
 * Java version (java -version)/platform:  1.8
```

**Issue**
The result provide by MaximumWeightBipartiteMatching.getMatching() is not correct if augmentation happens.

**Steps to reproduce (small coding example)**
create a simple bipart graph, whose partition1 consists of 2 nodes and parttion2 consists of 1 node, two edges are added with different weights (the first edges should be a little small). 

**Expected behaviour**
The returned matchweight should be the larger one.
However, in fact, the smaller one is returned.

**Other information**
I checked the source code of MaximumWeightBipartiteMatching (Version 1.2), and I found where the problem come:

In the method augmentPathTo():

       `while(var5.hasNext()) {
            e = var5.next();
            w = BigDecimal.valueOf(this.graph.getEdgeWeight((E) e));
            s = this.graph.getEdgeSource((E) e);
            t = this.graph.getEdgeTarget((E) e);
            this.matchedEdge.remove(s);
            this.matchedEdge.remove(t);
            **this.matchingWeight.subtract(w);**
            this.matching.remove(e);
        }

        var5 = free.iterator();

        while(var5.hasNext()) {
            e = var5.next();
            w = BigDecimal.valueOf(this.graph.getEdgeWeight((E) e));
            s = this.graph.getEdgeSource((E) e);
            t = this.graph.getEdgeTarget((E) e);
            this.matchedEdge.put(s, e);
            this.matchedEdge.put(t, e);
            **this.matchingWeight.add(w);**
            this.matching.add((E) e);
        }`

Here  **this.matchingWeight.subtract(w)** and  **this.matchingWeight.add(w)** do not actually update the value of matchingWeight

**Comment by d-michail** [comment]
Thanks. We have already fixed this. See our latest version.

Either use our latest release, or as a workaround recalculate the matching's weight after the algorithm returns.

**Comment by d-michail** [comment]
Fixed in #917
